### ISSUE TITLE
Correcting legacy "Unspecified" user types

### ISSUE THREAD

**Issue by Castronova** [hydroshare/hydroshare#3839]
Correct "Unspecifed" user types from existing user accounts. After #3812 is closed, this will no longer be an option and we'll need to fix any user accounts that have used it. "Unspecified" was removed because it wasn't informative with respect to NSF reporting.

Requirements:

1. Create a modal dialog that requires users to provide a new user type  if "Unspecified" was previously selected.
2. Prevent users from interacting with HS features until a new user type has been specified.

@jtpollak I tried to summarize your thoughts from #3812, please add corrections as necessary.

**Comment by jtpollak** [comment]
@Castronova this looks good! The modal dialog should pop-up as soon as a user authenticates.

**Comment by dtarb** [comment]
@Castronova @jtpollak do you want the dialog that pops up also to ask for Country and State/Province.  These are required fields now "unspecified" for many users.  Does the same logic need to apply to them for consistency.

Do we need any user messaging about this, i.e. warning to tell users that this is about to happen.  This will be a surprise and change the log in experience for many users as soon as it is deployed.  Is this what you want.

**Comment by jtpollak** [comment]
@hyi - @Castronova and I just discussed @dtarb comments and we want to make Country, State/Province required as well so it should have the same behavior as Unspecified user types.

**Comment by hyi** [comment]
@jtpollak Yes, I already made Country and State/Province required in addition to user type, and after my PR for correcting legacy users to ask them to input all missing required fields is reviewed and deployed, the user will be prompted to input any missing required fields including State/Province and Country.

**Comment by dtarb** [comment]
But accounts are being created today that do not have user type 
https://www.hydroshare.org/user/8106/ no user type (and promotional spam account)
https://www.hydroshare.org/user/8105/ no user type (and promotional spam account)
https://www.hydroshare.org/user/8104/ no user type (but did give title as “professor”)

I wonder if some browsers, or mobile configuration is allowing creation of accounts without giving a user account type.

**Comment by hyi** [comment]
@dtarb I am working on a fix to treat all whitespaces as empty for user types and state/province fields that allow users to type in anything they want. This is the only hole we found in the system that allows a user to get around the required constraint.

**Comment by hyi** [comment]
@Castronova I believe this issue can be closed.

**Comment by Castronova** [comment]
@hyi I believe that correcting existing user types has not yet been resolved.
